R/gt_univariate_stratified_internal.R

```diff
@@ -0,0 +1,409 @@
+
+
+## linelist_cleaned from test-gt_wrappers.R up to line 201 (dont worry this will be deleted later)
+# linelist_cleaned$gender_bin <- linelist_cleaned$gender == "Female"
+# linelist_cleaned$fever_bin  <- linelist_cleaned$fever == "Yes"
+# linelist_cleaned <- linelist_cleaned %>%
+#   mutate(obstime = sample.int(30, nrow(linelist_cleaned), replace = TRUE))
+
+
+
+tbl_cmh_internal <- function(data, case, exposure, strata, measure, obstime = NULL, conf.level = 0.95) {
+
+  ## make variables available for use
+  case_var <- tidyselect::vars_select(colnames(data), {{ case }})
+  exposure_var <- tidyselect::vars_select(colnames(data), {{ exposure }})
+  strata_var <- tidyselect::vars_select(colnames(data), {{ strata }})
+  obstime_var <- if (length(obstime) > 0) {
+    tidyselect::vars_select(colnames(data), {{ obstime }})} else NULL
+
+
+  ## if stratifying variable is not already a factor then make it one
+  if(!is.factor(data[[strata_var]])) {
+    data <- dplyr::mutate(data, {{strata}} := as.factor({{strata}}))
+  }
+
+
+  ## drop missings from all relevant variables
+  data <- tidyr::drop_na(data,
+                         {{case}},
+                         {{exposure}},
+                         {{strata}},
+                         {{obstime}})
+
+  ##### CRUDE ESTIMATES --------------------------------------------------------
+
+  ### TODO: simplify these by defining the args with switch() to pass to tbl_uvreg
+  if (measure == "OR") {
+    crude <- gtsummary::tbl_uvregression(data = data,
+                                  method = glm,
+                                  y = {{case}},
+                                  include = {{exposure}},
+                                  method.args = list(family = binomial),
+                                  exponentiate = TRUE,
+                                  hide_n = TRUE)
+  }
+
+  if (measure == "RR") {
+    crude <- gtsummary::tbl_uvregression(data = data,
+                                  method = glm,
+                                  y = {{case}},
+                                  include = {{exposure}},
+                                  method.args = list(family = poisson),
+                                  exponentiate = TRUE,
+                                  hide_n = TRUE)
+  }
+
+  if (measure == "IRR") {
+    crude <- gtsummary::tbl_uvregression(data = data,
+                                  method = glm,
+                                  y = {{case}},
+                                  include = {{exposure}},
+                                  method.args = list(family = poisson,
+                                                     offset = log({{obstime}})),
```

Comment:
chucking an error with {{obstime}} - cant figure out how to just feed all of this in to method.args clean - this would allow to specify up top and only need one call to gtsummary::tbl_uvregression

Reply:
note to self: create a temp var which is log ostime, then can just pass all as a char!